integration-libs/cdc/root/service/cdc-js.service.spec.ts

```diff
@@ -572,6 +582,233 @@ describe('CdcJsService', () => {
+            newPassword: newPass,
+            callback: jasmine.any(Function),
+          });
+        });
+      done();
+    });
+  });
+
+  describe('updateUserEmailWithoutScreenSet', () => {
+    it('should not call updateUserEmailWithoutScreenSet', (done) => {
+      spyOn(
+        winRef.nativeWindow['gigya'].accounts,
+        'setAccountInfo'
+      ).and.callFake((options: { callback: Function }) => {
+        options.callback({ status: 'OK' });
+      });
+      expect(service.updateProfileWithoutScreenSet).toBeTruthy();
+      service.updateUserEmailWithoutScreenSet('', '').subscribe(() => {
+        expect(
+          winRef.nativeWindow['gigya'].accounts.setAccountInfo
+        ).not.toHaveBeenCalled();
+      });
+      done();
+    });
+
+    it('should call updateUserEmailWithoutScreenSet', (done) => {
+      spyOn(
+        winRef.nativeWindow['gigya'].accounts,
+        'setAccountInfo'
+      ).and.callFake((options: { callback: Function }) => {
+        options.callback({ status: 'OK' });
+      });
+      spyOn(winRef.nativeWindow['gigya'].accounts, 'login').and.callFake(
+        (options: { callback: Function }) => {
+          options.callback({ status: 'OK' });
+        }
+      );
+
+      expect(service.updateUserEmailWithoutScreenSet).toBeTruthy();
+      let pass = 'Password123!';
+
+      service.updateUserEmailWithoutScreenSet(pass, newEmail).subscribe(() => {
+        expect(
+          winRef.nativeWindow['gigya'].accounts.setAccountInfo
+        ).toHaveBeenCalledWith({
+          profile: {
+            email: newEmail,
+          },
+          callback: jasmine.any(Function),
+        });
+        expect(userProfileFacade.update).toHaveBeenCalledWith({
+          uid: newEmail,
+        });
+      });
+      done();
+    });
+  });
+
+  describe('getLoggedInUserEmail', () => {
+    it('should return the logged in user email', () => {
+      userProfileFacade.get = createSpy().and.returnValue(
+        of({ uid: newEmail })
+      );
+
+      expect(service['getLoggedInUserEmail']()).toEqual(newEmail);
+    });
+
+    it('should return empty if no email is obtained', () => {
+      userProfileFacade.get = createSpy().and.returnValue(of({}));
+
+      expect(service['getLoggedInUserEmail']()).toEqual('');
+    });
+  });
+
+  describe('updateAddressWithoutScreenSet', () => {
+    it('should not call updateAddressWithoutScreenSet', (done) => {
+      spyOn(
+        winRef.nativeWindow['gigya'].accounts,
+        'setAccountInfo'
+      ).and.callFake((options: { callback: Function }) => {
+        options.callback({ status: 'OK' });
+      });
+      expect(service.updateProfileWithoutScreenSet).toBeTruthy();
+      service.updateAddressWithoutScreenSet('').subscribe(() => {
+        expect(
+          winRef.nativeWindow['gigya'].accounts.setAccountInfo
+        ).not.toHaveBeenCalled();
+      });
+      done();
+    });
+
+    it('should call updateAddressWithoutScreenSet', (done) => {
```

Comment:
Similar change suggestion applies to other places in this file:

```suggestion
    it('should call accounts.setAccountInfo', (done) => {
```